src/wallet/wallet.cpp

```diff
@@ -2381,9 +2381,62 @@ bool CWallet::SelectCoinsMinConf(const CAmount& nTargetValue, const CoinEligibil
     if (SelectCoinsBnB(positive_groups, nTargetValue, cost_of_change, setCoinsRet, nValueRet)) {
         return true;
     }
+
     // The knapsack solver has some legacy behavior where it will spend dust outputs. We retain this behavior, so don't filter for positive only here.
     std::vector<OutputGroup> all_groups = GroupOutputs(coins, !coin_selection_params.m_avoid_partial_spends, effective_feerate, long_term_feerate, eligibility_filter, false /* positive_only */);
-    return KnapsackSolver(nTargetValue + change_fee, all_groups, setCoinsRet, nValueRet);
+
+    std::set<CInputCoin> knapsack_coins;
+    CAmount knapsack_value;
+    bool knapsack_ret = KnapsackSolver(nTargetValue + change_fee, all_groups, knapsack_coins, knapsack_value);
+    CAmount knapsack_fees = 0;
+    for (const auto& coin : knapsack_coins) {
+        knapsack_fees += coin.m_fee;
+    }
+
+    std::set<CInputCoin> srd_coins;
+    CAmount srd_value;
+    bool srd_ret = SelectCoinsSRD(positive_groups, nTargetValue + change_fee + MIN_FINAL_CHANGE, srd_coins, srd_value);
```

Comment:
Would be nice if the order of the parameters on `KnapsackSolver` and `SelectCoinsSRD` matched.

Reply:
It would, but changes to `KnapsackSolver` are out of scope for this PR. I prefer to use the current order for `SelectCoinsSRD`.